src/SourceBrowser.Tests/CSharpTests.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.CodeAnalysis.UnitTests;
+using Microsoft.CodeAnalysis.UnitTests.SolutionGeneration;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using SourceBrowser.Generator.Model;
+using SourceBrowser.Generator;
+using SourceBrowser.Generator.Model.CSharp;
+
+namespace SourceBrowser.Tests
+{
+    [TestClass]
+    public class CSharpTests : MSBuildWorkspaceTestBase
+    {
+        [TestMethod]
+        public void SanityCheck()
+        {
+            //Set up the absolute minimum
+            var solution = base.Solution(
+                Project(
+                    ProjectName("Project1"),
+                    Sign,
+                    Document(
+                        @"
+                        class C1
+                        {
+                            public void M1 () { }
+                        }")));
+
+            WorkspaceModel ws = new WorkspaceModel("Workspace1", "");
+            FolderModel fm = new FolderModel(ws, "Project1");
+
+            var document = solution.Projects.SelectMany(n => n.Documents).Where(n => n.Name == "Document1.cs").Single();
+            var linkProvider = new ReferencesourceLinkProvider();
+
+            var walker = new SourceBrowser.Generator.DocumentWalkers.CSWalker(fm, document, linkProvider);
```

Comment:
var walker = SourceBrowser.Generator.DocumentWalkers.WalkerSelector.GetWalker(fm, document, linkProvider);

Also, make SourceBrowser.Generator.DocumentWalkers.WalkerSelector a public class
Apply this change to other tests